CONTRIBUTING.md

```diff
@@ -0,0 +1,54 @@
+
+## Commit messages
+1. The title: Use present tense imperative to shortly (less than 50 characters) describe your changes. You may reference an issue to further clarify which problem(s) your commit addresses.
+Furthermore, do not end the title with a period, as titles do not iend with periods. A valid title would be: `Bug #42: Fix camera API snapshot method`.
+2. Leave a blank line after your title.
+3. The body: Try to keep a column limit of around 50 characters to avoid horizontal scrolling. Use the body to further elaborate your changes using present tense imperative. You may name further TODOs if
+you referenced an issue and did not solve the problem completely. A fully valid commit message could look like this:
+`Bug #42: Fix camera API snapshot method
+
+Fix the method CameraAPI.snapshot() so the experimenter 
+can take single images. 
+TODO: 
+* Implement the CameraAPI.getImageStream() method
+* Document the API and write tests`.
+4. Remember to keep your commit messages as precise as possible.
+
+## Coding style
+Consistency is crucial to good collaboration between different contributors. For this purpose we enforce the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html).
+Please note that the XML specification of services results in the generation of Java classes and methods. Therefore, the names of services are supposed to be written as UpperCamelCase and the service 
+parameters are supposed to be written in lowerCamelCase.
+
+## Quality assurance (QA)
+The NMF is supposed to operate on Cubesats which are associated with a lot of work by dedicated teams. To ensure that their time and work is not wasted by erronous software, everyone contributing to the NMF shall try to keep the framework as usable and correct as possible.
+
+### Documentation
+Good documentation helps new users to implement their vision into code and also helps the reviewer to understand the intention behind your contribution. Furthermore, good documentation can be used to write
+tests and spot problems with your code. For the NMF we use Javadoc to document classes and methods.
+
+#### Classes
+1. Document your new classes with an @author tag, so we know who to contact if problems arise in the future.
+2. Briefly describe the purpose of your class.
+
+#### Methods
+1. Every method gets a Javadoc docstring. Make sure to use the first line for a brief summary (see [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html#s7-javadoc) for more 
+information).
+2. Describe precisely what your method accomplishes. 
+3. If its correctness depends on a certain state, state this behind an additional line `Precondition: STATE\_BEFORE\_EXECUTION` after the description. So testers can setup the test environment accordingly.
+4. If the method has side-effects, provide information about the new state of the program behind and additional line `Postcondition: STATE\_AFTER\_EXECUTION` which follows the precondition (if present).
```

Comment:
I am not entirely convinced enforcing Precondition/Postcondition documentation makes sense on a longer run and feels idealistic. Almost every method will have some pre/post conditions, and while it is useful for building a test sets to an existing code, I would dispute whether the added documenting and reading overhead is smaller than a slightly steeper learning curve.

Reply:
I think this comes down to personal preference. I would treat it like a DLC. It is not necessary but nice to have. So I will scratch it from the contribution guide.